src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1221,6 +1280,29 @@ AddNodeMetadata(char *nodeName, int32 nodePort,
 		return workerNode->nodeId;
 	}
 
+	if (nodeMetadata->groupId != COORDINATOR_GROUP_ID &&
```

Comment:
I initially thought that we may consider not allowing adding worker nodes irrespective of  `strcmp(nodeName, "localhost") != 0`. But, this approach makes testing locally much easier, so I'm on board now.